sample_app/app/controllers/users_controller.rb

```diff
@@ -0,0 +1,36 @@
+class UsersController < ApplicationController
+  before_action :find_user, only: :show
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new user_params
+
+    if @user.save
+      flash[:success] = t ".flash", username: @user.name
+      redirect_to user_path @user
+    else
+      render :new
+    end
+  end
+
+  def show
+    @user = User.find params[:id]
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit User::USERS_PARAMS
```

Comment:
in params, params.require(:user) ra đây cho anh?
Kiểu dữ liệu của chúng là gì?
Trình bày về Strong parameters

Reply:
- StrongParameters là 1 phương pháp để đối phó với lỗi bảo mật khi sử dụng tính năng Mass Assignment.
 - Mass Assignment là 1 tính năng cực kì tiện lợi cho phép update 1 model – khi tạo mới hoặc thay đổi – bằng hash của Ruby.
 - params: là một đối tượng của class ActionController::Parameters.